crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -43,39 +43,75 @@ impl From<AuthRpoFalcon512> for AccountComponent {
+/// 1. **Procedure-based authentication**: Requires authentication when any of the specified trigger
+///    procedures are called during the transaction.
+/// 2. **Output note authentication**: Controls whether creating output notes requires
+///    authentication. Output notes are new notes created by the account and sent to other accounts
+///    (e.g., when transferring assets). When `allow_unauthorized_output_notes` is `false`, any
+///    transaction that creates output notes must be authenticated, ensuring account owners control
+///    when their account sends assets to other accounts.
+/// 3. **Input note authentication**: Controls whether consuming input notes requires
+///    authentication. Input notes are notes that were sent to this account by other accounts (e.g.,
+///    incoming asset transfers). When `allow_unauthorized_input_notes` is `false`, any transaction
+///    that consumes input notes must be authenticated, ensuring account owners control when their
+///    account processes incoming notes.
 ///
-/// It exports the procedure `auth__tx_rpo_falcon512_procedure_acl`, which:
-/// - Checks if any of the specified auth trigger procedures were called during the transaction
-/// - If none were called, authentication is skipped
-/// - If at least one was called, performs standard RpoFalcon512 signature verification
-/// - Always increments the nonce
+/// ## Authentication Logic
 ///
-/// The storage layout is:
-/// - Slot 0(value): Public key (same as AuthRpoFalcon512)
-/// - Slot 1(value): Number of trigger procedures
+/// Authentication is required if ANY of the following conditions are true:
+/// - Any trigger procedure from the ACL was called
+/// - Output notes were created AND `allow_unauthorized_output_notes` is `false`
+/// - Input notes were consumed AND `allow_unauthorized_input_notes` is `false`
+///
+/// If none of these conditions are met, only the nonce is incremented without requiring a
+/// signature.
+///
+/// ## Use Cases
+///
+/// - **Restrictive mode** (`allow_unauthorized_output_notes=false`,
+///   `allow_unauthorized_input_notes=false`): All note operations require authentication, providing
+///   maximum security.
+/// - **Selective mode**: Allow some note operations without authentication while protecting
+///   specific procedures, useful for accounts that need to process certain operations
+///   automatically.
+/// - **Procedure-only mode** (`allow_unauthorized_output_notes=true`,
+///   `allow_unauthorized_input_notes=true`): Only specific procedures require authentication,
+///   allowing free note processing.
+///
+/// ## Storage Layout
+/// - Slot 0(value): Public key (same as RpoFalcon512)
+/// - Slot 1(value): [num_tracked_procs, allow_unauthorized_output_notes,
+///   allow_unauthorized_input_notes, 0]
 /// - Slot 2(map): A map with trigger procedure roots
 ///
 /// This component supports all account types.
 pub struct AuthRpoFalcon512Acl {
```

Comment:
It might be good to mention in the docs of this as well the caveats from https://github.com/0xMiden/miden-base/issues/1623, i.e. the fact that `was_procedure_called` will only return true if the procedure in question called into a kernel account API that is only allowed to be called from the account context. (This is quite complex and I wonder if anyone will understand this without knowing kernel internals...).